x-pack/solutions/observability/plugins/apm/server/routes/environments/route.ts

```diff
@@ -55,4 +55,28 @@ const environmentsRoute = createApmServerRoute({
   },
 });
 
-export const environmentsRouteRepository = environmentsRoute;
+const unsetEnvironmentsRoute = createApmServerRoute({
+  endpoint: 'GET /internal/apm/environments/unset',
```

Comment:
Could we run this query in the `GET /internal/apm/environments` endpoint?

Reply:
The `GET /internal/apm/environments` actually already handles Unset, but this query is only run on the services details. I suspect that the `useEnvironmentFetcher` uses `GET /internal/apm/suggestions` for the service inventory page for performance reasons. I could simplify it and use only `GET /internal/apm/environment` but I suspect it is separated out like this for a reason.

Reply:
Yes. you're right. I actually wanted to say `GET /internal/apm/suggestions`.Maybe you could run this second query in the `GET /internal/apm/suggestions` endpoint